openprovider/apis/params_creator.php

```diff
@@ -25,6 +27,43 @@ public function createParameters(array $args, $client, string $method): array
             $this->createParametersGetDelete($args, $client, $method);
     }
 
+    /**
+     * @param array $args
+     * @return array with formatted domain name by idn converter
+     */
+    public function modifyArgsIfDomainIdn(array $args): array
```

Comment:
why don't you encode domain extension?
I think you need to encode it also.

Reply:
I think it's better to move it method like idnToAscii of child classes for ParamsCreator e.g. CreateDomainParamsCreator